### PR TITLE
Draft verbiage for central security policy

### PR DESCRIPTION
**Pull request by kabir**

### CODE REVIEW

**Comment by fl4via** [comment]
@kabir I liked the idea! IMHO, it would be great to add as well something mentioning that the person needs to be careful to not submit PRs with the fix or reproducer, or even keep the reproducer/fix in a public repo, everything should be private at the moment of notification... something along these lines.

**Comment by kabir** [comment]
@fl4via I added another paragraph at the end

File: _layouts/security.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        <h2>Open a Jira issue</h2>
+        <p>If you are an experienced contributor, and choose to open a Jira issue in the correct project at
+            <a href="https://issues.redhat.com">https://issues.redhat.com</a>, make sure to set the
+            <strong>Security Level</strong> field to <strong>Embargoed Security Issue</strong>. This keeps it private.
+        </p>
+        <p>
+            Note that if you do open a Jira issue with the Security Level set to Embargoed Security Issue, you will not be able
+            to view the issue due to our security policy.
+        </p>
+    </div>
```
Comment (bstansberry) [inline _layouts/security.html:40]: I'm doubtful as to whether we should have this section. The simple policy a bunch of projects added only recommends email reporting, as does https://quarkus.io/security/ and https://www.keycloak.org/security.html.

If someone reads this and has two options it greatly increases the chances of them doing it wrong. If someone doesn't read this, it doesn't matter what it says. :)

File: _layouts/security.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            If you do do this, please make sure that you <strong>do not open any public pull requests</strong> since
+            that risks breaking the embargo.
+        </p>
+        <p>
+            Instead, make a private fork of the repository, and send us a message via <a href="mailto:XXX@redhat.com">XXX@redhat.com</a>.
+            A member of the team will be in touch. Once that happens, you will need to add them as a contributor to your private
+            fork so that they can see your changes.
+        </p>
+    </div>
+</div>
```
Comment (bstansberry) [inline _layouts/security.html:54]: I like what https://www.keycloak.org/security.html does about this, if we can work that out. It's simplest -- send us your info, we then get back to you on how to work the issue privately.

File: _layouts/security.html (hunk 1 above)
Comment (kabir) [reply _layouts/security.html:40]: @bstansberry I was in doubt whether to include this for the same reasons. I can remove it

File: _layouts/security.html (hunk 2 above)
Comment (kabir) [reply _layouts/security.html:43]: @bstansberry updated

File: _layouts/security.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+<div class="security-page grid-wrapper">
+    <div class="grid__item width-12-12">
+        <h1>Reporting of CVEs and Security Issues</h1>
+        <h2 class="page-subtitle">The Wild<strong>Fly</strong> community and our sponsor, Red Hat, take security bugs very
+            seriously
+        </h2>
+        <p>We aim to take immediate action to address serious security-related problems that involve our projects.
+            Note that we will only fix such issues in the most recent major release of Wild<strong>Fly</strong>.</p>
```
Comment (bstansberry) [inline _layouts/security.html:12]: I think "major" should be minor; e.g. we would not have fixed 26.0 once 26.1 was out.

File: _layouts/security.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        </h2>
+        <p>We aim to take immediate action to address serious security-related problems that involve our projects.
+            Note that we will only fix such issues in the most recent minor release of Wild<strong>Fly</strong>.</p>
+    </div>
+    <div class="grid__item width-12-12 sec-title-red">
+        <h2>Reporting of Security Issues</h2>
+        <p>When reporting a security vulnerability it is important to not accidentally broadcast to the world that the
+            issue exists, as this makes it easier for people to exploit it. The software industry uses the term
+            <a href="https://www.redhat.com/en/blog/security-embargoes-red-hat">embargo</a> to describe the time a security
+            issue is known internally until it public knowledge.
```
Comment (bstansberry) [inline _layouts/security.html:19]: s/it public/it is public/g

File: _layouts/security.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        </p>
+        <p>
+            Our preferred way of reporting security issues in Wild<strong>Fly</strong> and its related projects are listed below.
```
Comment (bstansberry) [inline _layouts/security.html:22]: s/are listed/is listed/g

File: _layouts/security.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    <div class="grid__item width-12-12 sec-title-yellow">
+        <h2>Other considerations</h2>
+        <p>
+            If you would like to work with us on a fix for the security vulnerability, please include your GitHub username
+            in the above email, and we will provide you access to a temporary private fork where we can collaborate on a
+            fix without it being disclosed publicly.
+        </p>
+        <p>
+            Do not open a public issue, send a pull request, or disclose any information about the suspected vulnerability
+            publicly. If you discover any publicly disclosed security vulnerabilities, please notify us immediately through
```
Comment (bstansberry) [inline _layouts/security.html:39]: s/publicly./publicly, including in your own publicly visible git repository./g

File: _layouts/security.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            <a href="https://www.redhat.com/en/blog/security-embargoes-red-hat">embargo</a> to describe the time a security
+            issue is known internally until it public knowledge.
+        </p>
+        <p>
+            Our preferred way of reporting security issues in Wild<strong>Fly</strong> and its related projects are listed below.
+        </p>
+    </div>
+    <div class="grid__item width-12-12 sec-title-teal">
+        <h2>Email the mailing list</h2>
+        <p>The list at <a href="mailto:XXX@redhat.com">XXX@redhat.com</a> is the preferred mechanism for outside users
```
Comment (bstansberry) [inline _layouts/security.html:27]: security@wildfly.org.  :)

File: _layouts/security.html [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        <h2>Other considerations</h2>
+        <p>
+            If you would like to work with us on a fix for the security vulnerability, please include your GitHub username
+            in the above email, and we will provide you access to a temporary private fork where we can collaborate on a
+            fix without it being disclosed publicly.
+        </p>
+        <p>
+            Do not open a public issue, send a pull request, or disclose any information about the suspected vulnerability
+            publicly. If you discover any publicly disclosed security vulnerabilities, please notify us immediately through
+            <a href="mailto:XXX@redhat.com">XXX@redhat.com</a>
```
Comment (bstansberry) [inline _layouts/security.html:40]: security@wildfly.org.  :)

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: _layouts/security.html (hunk 6 above)
Comment (bstansberry) [reply _layouts/security.html:39]: @kabir This one is still outstanding.

**Review by bstansberry — APPROVED** [review]

**Comment by kabir** [comment]
Screenshot 2023-06-16 at 11 04 59

Screenshot 2023-06-16 at 11 06 10

@bstansberry Screenshots of the rendered page (had to do two to get the full content)

**Comment by bstansberry** [comment]
Thanks @kabir and @fl4via

**Pull request merged**